rb/lib/selenium/server.rb

```diff
@@ -232,7 +232,11 @@ def stop_process
 
     def process
       @process ||= (
-        cp = ChildProcess.build('java', '-jar', @jar, '-port', @port.to_s, *@additional_args)
+        # extract any additional_args that start with -D as options
+        options = if @additional_args
+          @additional_args.dup - @additional_args.delete_if{|arg| arg.match(/^-D.*$/)}
```

Comment:
`@additional_args` is always going to return `true` since it defaults to `[]`
let's name it `properties` or `system_properties`, and we should avoid using `#match` where possible. Otherwise this is good. :)
I think my preference would be:
```
properties = @additional_args.dup - @additional_args.delete_if { |arg| arg[/^-D/] }
```